### PR TITLE
Add method to sign and confirm a meta-transaction to identity wallet

### PR DESCRIPTION
**Pull request by cducrest**
closes: https://github.com/trustlines-network/clientlib/issues/194

### CODE REVIEW

File: src/wallets/IdentityWallet.ts [hunk 1]
```diff
@@ -225,4 +317,27 @@ export class IdentityWallet implements TLWallet {
...
+      )
+    }
+  }
+
+  private buildMetaTransaction(rawTx: RawTxObject): MetaTransaction {
+    const metaTransaction: MetaTransaction = {
+      data: ethers.utils.solidityKeccak256(['bytes'], [rawTx.data]),
+      extraData: '0x',
+      from: rawTx.from,
+      nonce: rawTx.nonce.toString(),
```
Comment (berndbohmeier) [inline src/wallets/IdentityWallet.ts]: why does that have to be a string?

File: src/typings.ts [hunk 2]
```diff
@@ -208,6 +208,16 @@ export interface RawTxObject {
   nonce?: number
 }
 
+export interface MetaTransaction {
+  extraData: string
+  from: string
+  to: string
+  value: string
+  data: string
+  nonce: string
```
Comment (berndbohmeier) [inline src/typings.ts]: why is this a string?

File: src/wallets/IdentityWallet.ts [hunk 3]
```diff
@@ -186,6 +202,82 @@ export class IdentityWallet implements TLWallet {
...
+    ]
+    const values = [
+      '0x19',
+      '0x00',
+      metaTransaction.from,
+      metaTransaction.to,
+      metaTransaction.value,
+      ethers.utils.solidityKeccak256(['bytes'], [metaTransaction.data]),
+      metaTransaction.nonce
+    ]
```
Comment (berndbohmeier) [inline src/wallets/IdentityWallet.ts]: You forgot the extraData

File: tests/e2e/Identity.test.ts [hunk 4]
```diff
@@ -29,18 +33,49 @@ describe('e2e', () => {
...
+      it('should relay meta transaction and return a transaction hash', async () => {
+        const rawTx: RawTxObject = {
+          data: '0x',
+          from: identityWallet.address,
+          nonce: 1,
+          to: identityWallet.address,
+          value: 0
+        }
+
+        const transactionHash = await identityWallet.confirm(rawTx)
```
Comment (berndbohmeier) [inline tests/e2e/Identity.test.ts]: This is the transactionHash of the ethereum transaction? Is there a way to get the hash of the metaTransaction?

File: src/wallets/IdentityWallet.ts (hunk 1 above)
Comment (cducrest) [reply src/wallets/IdentityWallet.ts]: I can see in the relay server that it is interpreted in the `MetaTransactionSchema` as the same type as `value` (BigInteger) which afaict should be a string.

Comment (berndbohmeier) [reply src/wallets/IdentityWallet.ts]: Ah I am always confused that some of these types are types of the api. I'll create an issue to make that explicit.

**Pull request merged**